src/uu/shred/src/shred.rs

```diff
@@ -212,136 +213,164 @@ impl<'a> BytesGenerator<'a> {
+\n\
+* log-structured or journaled file systems, such as those supplied with\n\
+AIX and Solaris (and JFS, ReiserFS, XFS, Ext3, etc.)\n\
+\n\
+* file systems that write redundant data and carry on even if some writes\n\
+fail, such as RAID-based file systems\n\
+\n\
+* file systems that make snapshots, such as Network Appliance's NFS server\n\
+\n\
+* file systems that cache in temporary locations, such as NFS\n\
+version 3 clients\n\
+\n\
+* compressed file systems\n\
+\n\
+In the case of ext3 file systems, the above disclaimer applies\n\
+and shred is thus of limited effectiveness) only in data=journal mode,\n\
+which journals file data in addition to just metadata.  In both the\n\
+data=ordered (default) and data=writeback modes, shred works as usual.\n\
+Ext3 journaling modes can be changed by adding the data=something option\n\
+to the mount options for a particular file system in the /etc/fstab file,\n\
+as documented in the mount man page (man mount).\n\
+\n\
+In addition, file system backups and remote mirrors may contain copies\n\
+of the file that cannot be removed, and that will allow a shredded file\n\
+to be recovered later.\n\
+";
+
+pub mod options {
+    pub const FILE: &str = "file";
+    pub const ITERATIONS: &str = "iterations";
+    pub const SIZE: &str = "size";
+    pub const REMOVE: &str = "remove";
+    pub const VERBOSE: &str = "verbose";
+    pub const EXACT: &str = "exact";
+    pub const ZERO: &str = "zero";
+}
+
 pub fn uumain(args: impl uucore::Args) -> i32 {
     let args = args.collect_str();
 
-    let mut opts = getopts::Options::new();
-
-    // TODO: Add force option
-    opts.optopt(
-        "n",
-        "iterations",
-        "overwrite N times instead of the default (3)",
-        "N",
-    );
-    opts.optopt(
-        "s",
-        "size",
-        "shred this many bytes (suffixes like K, M, G accepted)",
-        "FILESIZE",
-    );
-    opts.optflag(
-        "u",
-        "remove",
-        "truncate and remove the file after overwriting; See below",
-    );
-    opts.optflag("v", "verbose", "show progress");
-    opts.optflag(
-        "x",
-        "exact",
-        "do not round file sizes up to the next full block; \
-         this is the default for non-regular files",
-    );
-    opts.optflag(
-        "z",
-        "zero",
-        "add a final overwrite with zeros to hide shredding",
-    );
-    opts.optflag("", "help", "display this help and exit");
-    opts.optflag("", "version", "output version information and exit");
-
-    let matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(e) => panic!("Invalid options\n{}", e),
+    let matches = App::new("shred")
+        .name("shred")
+        .version(VERSION_STR)
+        .about(
+            "Overwrite the specified FILE(s) repeatedly, in order to make it harder\n\
+for even very expensive hardware probing to recover the data.",
+        )
+        .after_help(AFTER_HELP)
+        .usage("shred [OPTION]... FILE...")
```

Comment:
please move that into a function like 
https://github.com/uutils/coreutils/blob/master/src/uu/ls/src/ls.rs#L50-L53 
for consistency with the other binaries